spynnaker/pyNN/models/neuron/abstract_population_vertex.py

```diff
@@ -251,15 +256,28 @@ def __init__(
 
         self.__structure = None
 
-    @overrides(TDMAAwareApplicationVertex.get_max_atoms_per_core)
+        # An RNG for use in synaptic generation
+        self.__rng = numpy.random.RandomState(seed)
+        self.__pop_seed = create_mars_kiss_seeds(self.__rng)
+        self.__core_seeds = dict()
+
+        # Store connections read from machine until asked to clear
+        # Key is app_edge, synapse_info
+        self.__connection_cache = dict()
+        self.__read_initial_values = False
```

Comment:
Should these not be set back to these values in reset_to_first_timestep

Reply:
Connection cache has an explicit clear, which is called before run.
Read initial values is set back to false after the values have been read.